src/encoder.js

```diff
@@ -65,8 +65,15 @@ function encodeString(str, target, lut) {
       target.push(tags.STR_);
       encodeUInt(numBytes, target);
     }
-    // todo: find way around blowing callstack with huge strings
-    target.push.apply(target, utf8Bytes);
+    const APPLY_CHUNK_SIZE = 0xFFFF;
+    if (utf8Bytes.length > APPLY_CHUNK_SIZE) {
+      while (utf8Bytes.length > 0) {
+        [].push.apply(target, utf8Bytes.slice(0, APPLY_CHUNK_SIZE));
+        utf8Bytes = utf8Bytes.slice(APPLY_CHUNK_SIZE);
```

Comment:
Laziness. Yours has fewer slices, so it'll probably be more performant. I'll use that.